module/VuFind/src/VuFind/Search/Summon/Options.php

```diff
@@ -67,91 +64,32 @@ class Options extends \VuFind\Search\Base\Options
     public function __construct(\VuFind\Config\PluginManager $configLoader)
     {
         $this->searchIni = $this->facetsIni = 'Summon';
+        $this->advancedFacetSettingsSection = 'Advanced_Facet_Settings';
         parent::__construct($configLoader);
-        // Load facet preferences:
-        if ($translatedFacets = $this->facetSettings['Advanced_Facet_Settings']['translated_facets'] ?? null) {
-            $this->setTranslatedFacets((array)$translatedFacets);
-        }
-        if ($specialFacets = $this->facetSettings['Advanced_Facet_Settings']['special_facets'] ?? null) {
-            $this->specialAdvancedFacets = $specialFacets;
-        }
-
-        // Load the search configuration file:
-        $searchSettings = $configLoader->get($this->searchIni);
-
-        // Set up limit preferences
-        if (isset($searchSettings->General->default_limit)) {
-            $this->defaultLimit = $searchSettings->General->default_limit;
-        }
-        if (isset($searchSettings->General->limit_options)) {
-            $this->limitOptions = $this->explodeListSetting($searchSettings->General->limit_options);
-        }
 
         // Set up highlighting preference
-        if (isset($searchSettings->General->highlighting)) {
-            $this->highlight = $searchSettings->General->highlighting;
+        if (null !== ($highlighting = $this->searchSettings['General']['highlighting'] ?? null)) {
+            $this->highlight = $highlighting;
         }
 
         // Set up spelling preference
-        if (isset($searchSettings->Spelling->enabled)) {
-            $this->spellcheck = $searchSettings->Spelling->enabled;
+        if (null !== ($spellcheck = $this->searchSettings['Spelling']['enabled'] ?? null)) {
+            $this->spellcheck = $spellcheck;
         }
 
-        // Load search preferences:
-        if (isset($searchSettings->General->default_filters)) {
-            $this->defaultFilters = $searchSettings->General->default_filters
-                ->toArray();
-        }
-        if (isset($searchSettings->General->result_limit)) {
-            $this->resultLimit = $searchSettings->General->result_limit;
-        } else {
-            $this->resultLimit = 400;   // default
+        // Set default result limit to a value we can support:
+        if (-1 === $this->resultLimit) {
```

Comment:
Maybe a `defaultResultLimit` property would be a more stable way to achieve this, but I don't mind leaving as-is if you prefer.

Reply:
Added. I also added `maxResultLimit` for the ProQuestFSG case.